build/xtask/src/dist.rs

```diff
@@ -737,6 +737,73 @@ fn build_archive(cfg: &PackageConfig, image_name: &str) -> Result<()> {
         )?;
     }
 
+    //
+    // Iterate over tasks looking for elements that should be copied into
+    // the archive.  These are specified by the "copy-to-archive" array,
+    // which consists of keys into the config table; the values associated
+    // with these keys have the names of the files to add to the archive.
+    // All files added to the archive for a particular task will be in
+    // a directory dedicated to that task; all such directories will
+    // themselves be subdirectories in the "task" directory.
+    //
+    for (name, task) in &cfg.toml.tasks {
+        for c in &task.copy_to_archive {
+            match &task.config {
+                None => {
+                    bail!(
+                        "task {name}: {c} is specified to be copied \
+                        into archive, but config table is missing"
+                    );
+                }
+                Some(config) => match config.get(c) {
+                    Some(ordered_toml::Value::String(s)) => {
+                        //
+                        // This is a bit of a heavy hammer:  we need the
+                        // directory name for the task to find the file to be
+                        // copied into the archive, so we're going to iterate
+                        // over all packages to find the crate assocated with
+                        // this task.
+                        //
+                        use cargo_metadata::MetadataCommand;
```

Comment:
Since this is so expensive, how about caching it, e.g. putting
```rust
let mut metadata = None;
for (name, task) in &cfg.toml.tasks {
    // ....
    let metadata = match metadata.as_ref() {
        Some(m) => m,
        None => {
            let d = MetadataCommand::new()
                .manifest_path("./Cargo.toml")
                .exec()?;
            metadata.get_or_insert(d)
        }
    };
```

Reply:
Yeah, that's a good idea. I went so far as to clock it (~160 ms), and decided that caching wasn't practical -- but it's just too easy to do and too offensive to not do.